src/amp-story-player-impl.js

```diff
@@ -66,29 +66,31 @@ export const IFRAME_IDX = '__AMP_IFRAME_IDX__';
  * Note that this is a vanilla JavaScript class and should not depend on AMP
  * services, as v0.js is not expected to be loaded in this context.
  */
-export class AmpStoryPlayer {
+export class AmpStoryPlayer extends HTMLElement {
   /**
-   * @param {!Window} win
-   * @param {!Element} element
    * @constructor
    */
-  constructor(win, element) {
-    console./*OK*/ assert(
-      element.childElementCount > 0,
-      'Missing configuration.'
-    );
+  constructor() {
+    super();
+  }
+
+  /**
+   * Player is appended in the document.
+   */
+  connectedCallback() {
+    console./*OK*/ assert(this.childElementCount > 0, 'Missing configuration.');
 
     /** @private {!Window} */
-    this.win_ = win;
+    this.win_ = self;
 
     /** @private {!Array<!Element>} */
     this.iframes_ = [];
 
     /** @private {!Element} */
-    this.element_ = element;
+    this.element_ = this;
```

Comment:
Why would you need to store this as a property?

Reply:
Muscle memory 🤷🏽‍♂️ removed it